util/llpcPipelineDumper.cpp

```diff
@@ -908,16 +908,35 @@ void PipelineDumper::DumpGraphicsPipelineInfo(
 // Builds hash code from graphics pipline build info.
 MetroHash::Hash PipelineDumper::GenerateHashForGraphicsPipeline(
     const GraphicsPipelineBuildInfo* pPipeline,   // [in] Info to build a graphics pipeline
-    bool                            isCacheHash   // TRUE if the hash is used by shader cache
+    bool                            isCacheHash,  // TRUE if the hash is used by shader cache
+    uint32_t                        stage
     )
 {
     MetroHash64 hasher;
 
-    UpdateHashForPipelineShaderInfo(ShaderStageVertex, &pPipeline->vs, isCacheHash, &hasher);
-    UpdateHashForPipelineShaderInfo(ShaderStageTessControl, &pPipeline->tcs, isCacheHash, &hasher);
-    UpdateHashForPipelineShaderInfo(ShaderStageTessEval, &pPipeline->tes, isCacheHash, &hasher);
-    UpdateHashForPipelineShaderInfo(ShaderStageGeometry, &pPipeline->gs, isCacheHash, &hasher);
-    UpdateHashForPipelineShaderInfo(ShaderStageFragment, &pPipeline->fs, isCacheHash, &hasher);
+    switch (stage) {
+      case ShaderStageVertex:
+        UpdateHashForPipelineShaderInfo(ShaderStageVertex, &pPipeline->vs, isCacheHash, &hasher);
+        break;
+      case ShaderStageTessControl:
+        UpdateHashForPipelineShaderInfo(ShaderStageTessControl, &pPipeline->tcs, isCacheHash, &hasher);
+        break;
+      case ShaderStageTessEval:
+        UpdateHashForPipelineShaderInfo(ShaderStageTessEval, &pPipeline->tes, isCacheHash, &hasher);
+        break;
+      case ShaderStageGeometry:
+        UpdateHashForPipelineShaderInfo(ShaderStageGeometry, &pPipeline->gs, isCacheHash, &hasher);
+        break;
+      case ShaderStageFragment:
+        UpdateHashForPipelineShaderInfo(ShaderStageFragment, &pPipeline->fs, isCacheHash, &hasher);
+        break;
+      default:
+        UpdateHashForPipelineShaderInfo(ShaderStageVertex, &pPipeline->vs, isCacheHash, &hasher);
+        UpdateHashForPipelineShaderInfo(ShaderStageTessControl, &pPipeline->tcs, isCacheHash, &hasher);
+        UpdateHashForPipelineShaderInfo(ShaderStageTessEval, &pPipeline->tes, isCacheHash, &hasher);
+        UpdateHashForPipelineShaderInfo(ShaderStageGeometry, &pPipeline->gs, isCacheHash, &hasher);
+        UpdateHashForPipelineShaderInfo(ShaderStageFragment, &pPipeline->fs, isCacheHash, &hasher);
+    }
```

Comment:
I am afraid something wrong in this function. it always add all pipeline state after the switch case regardless which shader stage is specified. please double check it.

Reply:
Yes, I could do better here.  It would still work, but not as well as it could.  I'll update it.